src/crepe/api/Sprite.h

```diff
@@ -42,25 +41,42 @@ class Sprite : public Component {
 	 * \param image Shared pointer to the texture for this sprite.
 	 * \param color Color tint applied to the sprite.
 	 * \param flip Flip settings for horizontal and vertical orientation.
+	 * \param order_layer decides the sorting in layer of the sprite.
+	 * \param sort_layer decides the order in layer of the sprite.
+	 * \param height the height of the image in game units
 	 */
-	Sprite(game_object_id_t id, const std::shared_ptr<Texture> image, const Color & color,
-		   const FlipSettings & flip);
+	Sprite(game_object_id_t id, const Texture & image, const Color & color,
+		   const FlipSettings & flip, uint8_t sort_layer, uint8_t order_layer, int height);
 
 	/**
 	 * \brief Destroys the Sprite instance.
 	 */
 	~Sprite();
 
 	//! Texture used for the sprite
-	const std::shared_ptr<Texture> sprite_image;
+	const Texture & sprite_image;
+
 	//! Color tint of the sprite
 	Color color;
+
 	//! Flip settings for the sprite
 	FlipSettings flip;
+
 	//! Layer sorting level of the sprite
-	uint8_t sorting_in_layer = 0;
+	const uint8_t sorting_in_layer;
 	//! Order within the sorting layer
-	uint8_t order_in_layer = 0;
+	const uint8_t order_in_layer;
+
+	//! height in world units
+	const int height;
+
+	/**
+	 * \aspect_ratio ratio of the img so that scaling will not become weird
+	 *
+	 * cannot be const because if Animator component is addded then ratio becomes scuffed and
+	 * does it need to be calculated again in the Animator
+	 */
+	double aspect_ratio;
 
 public:
 	/**
```

Comment:
Waarom is get_instances_max hieronder override? Volgens mij is er geen limiet op Sprites omdat er ook geen limiet is aan Animators. Dus get_instances_max moet -1 returnen (en dat gebeurt standaard al als je niet override).